### PR TITLE
Cleanup efs:install_utils

### PR DESCRIPTION
**Pull request by awslitvit**
### Description of changes
* Cleanup efs:install_utils

### Tests
* ChefSpec
* Kitchen test for `efs` manually run on EC2

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-common/resources/efs/partial/_mount_umount.rb [hunk 1]
```diff
@@ -18,8 +18,6 @@
 property :efs_encryption_in_transit_array, Array, required: false
 property :efs_iam_authorization_array, Array, required: false
 
-default_action :mount
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-common/resources/efs/partial/_mount_umount.rb:21]: Why did you remove the default action?
are we sure we're calling it always with an action specified?

Comment (awslitvit) [reply cookbooks/aws-parallelcluster-common/resources/efs/partial/_mount_umount.rb:21]: Yes, the default action is `:install_utils`, this was actually a bug and it made `efs:install_utils` test fail on EC2.

Comment (enrico-usai) [reply cookbooks/aws-parallelcluster-common/resources/efs/partial/_mount_umount.rb:21]: so I'd add:
```
default_action :install_utils
```
what do you think? This is aligned with all the other resources on which we're always defining a default action.

File: cookbooks/aws-parallelcluster-common/resources/efs/partial/_debian.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+def install_script_code(efs_utils_tarball, efs_utils_package, efs_utils_version)
+  <<-EFSUTILSINSTALL
+      set -e
+      tar xf #{efs_utils_tarball}
+      cd efs-utils-#{efs_utils_version}
+      ./build-deb.sh
+      apt-get -y install ./build/#{efs_utils_package}*deb
+  EFSUTILSINSTALL
+end
```
Comment (francesco-giordano) [inline cookbooks/aws-parallelcluster-common/resources/efs/partial/_debian.rb:24]: What about parametrise `./build-deb.sh`, `apt-get` and `*deb` to not have to difference here and in cookbooks/aws-parallelcluster-common/resources/efs/partial/_redhat_based.rb?

**Review by francesco-giordano — APPROVED** [review]

File: cookbooks/aws-parallelcluster-common/resources/efs/partial/_debian.rb (hunk 2 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-common/resources/efs/partial/_debian.rb:24]: I thought about it and discarded the option to keep the script flexible and future-proof.
